tdrs-backend/tdpservice/parsers/util.py

```diff
@@ -86,3 +87,221 @@ def contains_encrypted_indicator(line, encryption_field):
+    """Decrement the input quarter text by one."""
+    array = [1, 2, 3, 4]  # wrapping around an array
+    int_qtr = int(fiscal_quarter[1:])  # remove the 'Q', e.g., 'Q1' -> '1'
+    if int_qtr == 1:
+        year = year - 1
+
+    ind_qtr = array.index(int_qtr)  # get the index so we can easily wrap-around end of array
+    return year, "Q{}".format(array[ind_qtr - 1])  # return the previous quarter
+
+
+def transform_to_months(quarter):
+    """Return a list of months in a quarter."""
+    match quarter:
+        case "Q1":
+            return ["Jan", "Feb", "Mar"]
+        case "Q2":
+            return ["Apr", "May", "Jun"]
+        case "Q3":
+            return ["Jul", "Aug", "Sep"]
+        case "Q4":
+            return ["Oct", "Nov", "Dec"]
+        case _:
+            raise ValueError("Invalid quarter value.")
+
+def month_to_int(month):
+    """Return the integer value of a month."""
+    return datetime.strptime(month, '%b').month
+
+
+def case_aggregates_by_month(df, dfs_status):
+    """Return case aggregates by month."""
+    section = str(df.section)  # section -> text
+    program_type = get_prog_from_section(section)  # section -> program_type -> text
+
+    # from datafile year/quarter, generate short month names for each month in quarter ala 'Jan', 'Feb', 'Mar'
+    calendar_year, calendar_qtr = fiscal_to_calendar(df.year, df.quarter)
+    month_list = transform_to_months(calendar_qtr)
+
+    short_section = get_text_from_df(df)['section']
+    schema_models_dict = get_program_models(program_type, short_section)
+    schema_models = [model for model in schema_models_dict.values()]
+
+    aggregate_data = {}
+    for month in month_list:
+        total = 0
+        rejected = 0
+        accepted = 0
+        month_int = month_to_int(month)
+        rpt_month_year = int(f"{calendar_year}{month_int}")
+
+        if dfs_status == "Rejected":
+            # we need to be careful here on examples of bad headers or empty files, since no month will be found
+            # but we can rely on the frontend submitted year-quarter to still generate the list of months
+            aggregate_data[month] = {"accepted": "N/A", "rejected": "N/A", "total": "N/A"}
+            continue
+
+        case_numbers = set()
+        for schema_model in schema_models:
+            if isinstance(schema_model, SchemaManager):
+                schema_model = schema_model.schemas[0]
+
+            curr_case_numbers = set(schema_model.model.objects.filter(datafile=df).filter(RPT_MONTH_YEAR=rpt_month_year)
+                                    .distinct("CASE_NUMBER").values_list("CASE_NUMBER", flat=True))
+            case_numbers = case_numbers.union(curr_case_numbers)
+
+        total += len(case_numbers)
+        rejected += ParserError.objects.filter(case_number__in=case_numbers).distinct('case_number').count()
+
+        accepted = total - rejected
+
+        aggregate_data[month] = {"accepted": accepted, "rejected": rejected, "total": total}
```

Comment:
the new columns are `Accepted Without Errors`, `Accepted With Errors`, and `Rows Unable to Process` - i think the algorithm needs to be updated to reflect this too. The temporary data structure i came up with for #2643 is something like this

`case_aggregates`
```json
{
    "rejected": 1,
    "months": [
        {
            "month": "January",
            "accepted_without_errors": 50,
            "accepted_with_errors": 10,
        }
    ]
}
```

this also reflects a suggestion to make the `months` into an array with a `month` string, rather than an object with the month as a key - the object approach has the potential to be unordered when stored in the db and parsed on the frontend.

Reply:
Perhaps I misunderstood -- are these changes not being introduced in your PR?

Reply:
Ok, I will update the structure -- minor note, the months will be the 3-char abbreviated form.